azkaban-common/src/main/java/azkaban/event/EventHandler.java

```diff
@@ -41,7 +44,12 @@ public void fireEventListeners(final Event event) {
     final ArrayList<EventListener> listeners =
         new ArrayList<>(this.listeners);
     for (final EventListener listener : listeners) {
-      listener.handleEvent(event);
+      try {
+        listener.handleEvent(event);
+      } catch (Exception e) {
+        logger.warn("Error while calling handleEvent for: " + listener.getClass());
+        logger.warn(e.getMessage(), e);
+      }
```

Comment:
Wondering what's the benefit of catching java.lang.Exceptions for EventListener.handleEvent() when handleEvent doesn't declare a throw in its signature:

```
public interface EventListener {
  public void handleEvent(Event event);
}
```
I am afraid that this might be dead code, that may never get executed.

Reply:
It catches perhaps Unchecked exceptions, which would help if there are any Runtime exceptions. 
We dont want other listeners to not get impacted irrespective of failures with one. 
(It does not covers Error, so my question is on the flip: why not catch Throwable?) :)

Reply:
@anish-mal  - @abti is correct. This will catch unchecked exceptions. This code path was tested by:
1. Keeping the existing bug and making sure rest of the event handlers were executed.
2. Throwing an exception from a different event handler (JobCallbackManager) and making sure the rest of the handlers were called.

Reply:
I feel let's catch java.lang.RuntimeException and java.lang.Error explicitly. This makes the code itself very clear of what our intention is.

Reply:
The current code will not catch  java.lang.Error.

Reply:
Yes, not sure about catching  java.lang.Error - you are right. So let's leave that aside while we ponder over this. But without loss of what we are trying - java.lang.RuntimeException will clearly indicate our intention. Please consider doing this.